### PR TITLE
Adds stricter unittest of Isochrones

### PR DESCRIPTION
**Pull request by PureW**
Yes, hardcoding an expected response from Isochrones is
not ideal but it is at least much better then the current
test that basically checks if Isochrones returned succesfully.

Given the coming refactorings to Isochrones, this seems like
a good first sanity check.

### CODE REVIEW

File: test/isochrone.cc [hunk 1]
```diff
@@ -94,16 +103,27 @@ void test_isochrones() {
...
-      R"({"locations":[{"lat":52.078937,"lon":5.115321}],"costing":"auto","contours":[{"time":15}],"polygons":false,"denoise":0.2,"generalize":150})";
-  try_isochrone(reader, loki_worker, thor_worker, test_request1, expected1);
-
-  // Try pedestrian isochrone with one contour, polygon=true
-  std::string expected2 = "\"type\":\"Polygon\"";
-  const auto test_request2 =
-      R"({"locations":[{"lat":52.078937,"lon":5.115321}],"costing":"bicycle","contours":[{"time":15}],"polygons":true,"denoise":0.2})";
-  try_isochrone(reader, loki_worker, thor_worker, test_request2, expected2);
+  {
+    std::cout << "test-case 1" << std::endl;
```
Comment (danpaz) [inline test/isochrone.cc]: Cleanup debug statements

File: test/isochrone.cc [hunk 2]
```diff
@@ -94,16 +103,27 @@ void test_isochrones() {
 // Test auto isochrone with one contour
 // 32bit builds fail in release mode we'll look at this separately
 #if _WIN64 || __amd64__
```
Comment (danpaz) [inline test/isochrone.cc]: > // 32bit builds fail in release mode we'll look at this separately

Is this still the case?

File: test/isochrone.cc (hunk 1 above)
Comment (PureW) [reply test/isochrone.cc]: This was actually helpful when the test failed, it's not printed if test passes.

File: test/isochrone.cc [hunk 3]
```diff
@@ -94,16 +103,27 @@ void test_isochrones() {
...
-  const auto test_request1 =
-      R"({"locations":[{"lat":52.078937,"lon":5.115321}],"costing":"auto","contours":[{"time":15}],"polygons":false,"denoise":0.2,"generalize":150})";
-  try_isochrone(reader, loki_worker, thor_worker, test_request1, expected1);
-
-  // Try pedestrian isochrone with one contour, polygon=true
-  std::string expected2 = "\"type\":\"Polygon\"";
-  const auto test_request2 =
-      R"({"locations":[{"lat":52.078937,"lon":5.115321}],"costing":"bicycle","contours":[{"time":15}],"polygons":true,"denoise":0.2})";
-  try_isochrone(reader, loki_worker, thor_worker, test_request2, expected2);
+  {
```
Comment (mandeepsandhu) [inline test/isochrone.cc]: Just curious, why are these within a scope?

File: test/isochrone.cc (hunk 1 above)
Comment (mandeepsandhu) [reply test/isochrone.cc]: > This was actually helpful when the test failed, it's not printed if test passes.

Shouldn't that be captured via a assert message (indicating what failed)?

File: test/isochrone.cc (hunk 3 above)
Comment (PureW) [reply test/isochrone.cc]: The smaller scopes, the less likely you have unintended side effects. It visually conveys pretty strongly too that there are two test-cases without having to do a bunch of plumbing with new functions etc.

File: test/isochrone.cc (hunk 1 above)
Comment (PureW) [reply test/isochrone.cc]: We're currently trying to switch to gtest (https://github.com/valhalla/valhalla/pull/2128 is 99% there) so there wasn't much point in using existing asserts when all of that goes away when #2128 is merged.

Comment (mandeepsandhu) [reply test/isochrone.cc]: Can you add a little more description to this `cout` statement, similar to what you have put in the next test following this one?

File: test/isochrone.cc (hunk 3 above)
Comment (mandeepsandhu) [reply test/isochrone.cc]: Or are these multiple steps of the same test?

File: test/isochrone.cc (hunk 2 above)
Comment (PureW) [reply test/isochrone.cc]: Yup, segmentation fault on 32bit x86... https://circleci.com/gh/valhalla/valhalla/17069

Comment (PureW) [reply test/isochrone.cc]: Opened issue https://github.com/valhalla/valhalla/issues/2187

**Comment by PureW** [comment]
I rebased and adjusted the test for gtest.

File: test/isochrone.cc (hunk 1 above)
Comment (PureW) [reply test/isochrone.cc]: I ended up removing this line.

File: test/isochrone.cc [hunk 4]
```diff
@@ -69,16 +69,26 @@ void try_isochrone(GraphReader& reader,
...
   loki_worker.isochrones(request);
 
   // Process isochrone request
-  auto result = thor_worker.isochrones(request);
+  auto response_json = thor_worker.isochrones(request);
 
-  // Check if the result contains the expected string
-  EXPECT_NE(result.find(expected), std::string::npos) << "isochrones failed: expected " + expected;
+  rapidjson::Document response;
+  response.Parse(response_json);
```
Comment (mandeepsandhu) [inline test/isochrone.cc]: Bit confusing that `response_json` is a string but `response` is a json object. I'll leave it to you if you want to change. Rest of it LGTM! 👍

**Pull request merged**